activerecord/test/cases/adapters/postgresql/json_test.rb

```diff
@@ -192,6 +192,20 @@ def test_changes_in_place
 
     assert_equal({ "one" => "two", "three" => "four" }, json.payload)
     assert_not json.changed?
+
+    json.payload = { "three" => "four", "one" => "two" }
```

Comment:
This file is already mixed into both JSON & JSONB test suites, it looks like the MySQL json_test file is mostly a copy-paste of this one.

I've added a couple more commits, the first adds the same test into the MySQL tests as well.
The second extracts out all the shared tests into a shared module for both MySQL and PostgreSQL